### PR TITLE
feat: algorand accounts

### PR DESCRIPTION
**Pull request by bwmx**
## Description

Migration of an internal algorand-accounts extension upstream.

An issue detected with an edge case of subscriber timing out.. 

- the subscriber behavior now gracefully handles errors/retries (separate unit tests)
- log store (optional) is used instead of console.* logging
- now the algorand address is as used in the `address` property of the `Account` instead of base64 version previously.
- the subscriber now polls instead of using background (managed) polling, abortcontroller (used internally) throws DOMException which we can't handle gracefully in react native environment,

## TODO

- [x] migrate unit tests over
- [x] use logstore 
- [x] handle edge case where subscriber fails/dies and restart gracefully with timeout/retry
- [x] replace console.log statement in `createSubscriberWithWatchlist` with appropriate `onError` optional callback
- [x] use correct types on `WithAlgorandAccounts` 
- [x] add to example app
- [x] demo the algorand accounts/subscriber behavior in example app

### CODE REVIEW

**Review by PhearZero — COMMENTED** [review]
This is looking good, can we wire this into the example app? The example app only supports keystore-accounts, it will likely need some changes to generate other accounts.

File: accounts/algorand-extension/package.json [hunk 1]
```diff
@@ -0,0 +1,61 @@
+{
+  "name": "@algorandfoundation/accounts-algorand-extension",
```
Comment (PhearZero) [inline accounts/algorand-extension/package.json]: Very small nit:

`@algorandfoundation/algorand-accounts-extension`

It will be easier to match if we move the `algorand` as a prefix. (Select all "algorand" packages). Didn't catch this before

**Review by PhearZero — COMMENTED** [review]
I am going to run though this in a few, I've requested the package publication and will approve/merge after QA and package is available
